cmd/cluster/core/destroy.go

```diff
@@ -129,17 +131,41 @@ func DestroyCluster(ctx context.Context, hostedCluster *hyperv1.HostedCluster, o
 				o.Log.Error(err, "Failed to get hosted cluster", "namespace", o.Namespace, "name", o.Name)
 				return false, nil
 			}
-			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == destroyFinalizer
+			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == DestroyClusterFinalizer
 			return done, nil
 		}, clusterDeleteCtx.Done())
 		if err != nil {
 			return fmt.Errorf("hostedcluster wasn't finalized, aborting delete: %w", err)
 		}
 	}
 
-	// Destroy additional resources which are specific to the current platform
-	if err := destroyPlatformSpecifics(ctx, o); err != nil {
-		return err
+	// Destroy additional resources which are specific to the current platform (Skip and just remove finalizer with HCI)
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: corev1.ObjectMeta{
+			Name:      o.InfraID,
+			Namespace: o.Namespace,
+		},
+	}
+	hciFound := false
+	if err := c.Get(ctx, client.ObjectKeyFromObject(&hci), &hci); err == nil {
+		hciFound = true
+		//Remove finalizer on HCI
+		origHCI := hci.DeepCopy()
+		controllerutil.RemoveFinalizer(&hci, DestroyClusterFinalizer)
+		if err := c.Patch(ctx, &hci, client.MergeFrom(origHCI)); err != nil {
+			return fmt.Errorf("failed to remove finalizer: %w", err)
+		}
+		o.Log.Info("Found Hosted cluster infrastructure",
+			"namespace", o.Namespace, "name", o.InfraID)
```

Comment:
What's the relationship between HCI and HostedClusters? Is it possible that one HCI can map to multiple HostedClusters? If so, this finalizer logic might not be safe.

Is there anyway the HCI controller could handle this book keeping, and ensure that the HCI isn't removed if any HostedCluster exists that references it?